keps/sig-node/3673-kubelet-parallel-image-pull-limit/README.md

```diff
@@ -0,0 +1,860 @@
+Currently neither kubelet or containerd limits the number of inflight image pulls.
+
+On kubelet, as mentioned above, the QPS/burst limit does not take the in-progress pulls into account. On containerd, there is only [a per-image limit](https://github.com/containerd/containerd/blob/8e787543deede10f372b0e16ad5c07790fc680b6/pkg/cri/config/config.go#L299-L300), which only limits the number of parallel layer downloading for each image, but potentially allows unlimited number downloads overall.
+
+On containerd, [MaxConcurrentDownloads](https://github.com/containerd/containerd/blob/f6232793b49730435aa7fbc689c717efd35e0b28/pkg/cri/config/config.go#L300) limits the number of parallel downloads per image, but does not limit the number of downloads overall either.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+Adding a node-level limit of parallel image pulls to kubelet. This limit will limits the maximum number of images being pulled in parallel. Any image pull request beyond the limit will be blocked until one image pull finishes.
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Prioritizing image pulls in any way.
+* Using the number of inflight image pulls as a signal to direct pod scheduling.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+* A new integer field `maxParallelImagePulls` will be added to Kubelet Configuration. `maxParallelImagePuls` is the maximum number of inflight image pulls.
+* The default value of `maxParallelImagePulls` will be set to 5, which is the same as the default `--registry-qps`.
+* When `serialize-image-pulls` is true, `maxParallelImagePulls` will be ignored. A warning message will be logged by kubelet if `serialize-image-pulls` is true and `maxParallelImagePulls` is larger than 1.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+A kubernetes user wants to enable parallel image pulls to increase workload startup latency, but they don't want too many images to be pulled at the same time, as that might burden the disk IO and has a negative impact on disk performance. By setting a proper `maxParallelImagePulls`, they will have a better control over the parallelism and therefore avoid overwhelming the disk.
```

Comment:
> A kubernetes user wants to enable parallel image pulls to increase workload startup latency

Should this not be `A kubernetes user wants to enable parallel image pulls to decrease workload startup latency`

Reply:
thanks for catching it!